crates/runtime/src/lib.rs

```diff
@@ -609,7 +614,7 @@ impl Runtime {
                     .update_tool(&tool.name, ComponentStatus::Initializing);
             }
 
-            for tool_catalog in default_available_catalogs() {
+            for tool_catalog in default_available_catalogs(Arc::clone(&self)) {
```

Comment:
It's a bit inefficient to construct the entire tool just to get its name. Let's create a new method that just returns the default catalog names and use that?